com.unity.ml-agents/Runtime/EnvironmentParameters.cs

```diff
@@ -0,0 +1,70 @@
+using System;
+using System.Collections.Generic;
+using MLAgents.SideChannels;
+
+namespace MLAgents
+{
+    /// <summary>
+    /// A container for the Environment Parameters that may be modified during training.
+    /// The keys for those parameters are defined in the trainer configurations and the
+    /// the values are generated from the training process in features such as Curriculum Learning
+    /// and Environment Parameter Randomization.
+    ///
+    /// One current assumption for all the environment parameters is that they are of type float.
+    /// </summary>
+    public sealed class EnvironmentParameters
+    {
+        /// <summary>
+        /// The side channel that is used to receive the new parameter values.
+        /// </summary>
+        readonly EnvironmentParametersChannel m_Channel;
+
+        /// <summary>
+        /// Constructor.
+        /// </summary>
+        internal EnvironmentParameters()
+        {
+            m_Channel = new EnvironmentParametersChannel();
+            SideChannelsManager.RegisterSideChannel(m_Channel);
+        }
+
+        /// <summary>
+        /// Returns the parameter value for the specified key. Returns the default value provided
+        /// if this parameter key does not have a value. Only returns a parameter value if it is
+        /// of type float.
+        /// </summary>
+        /// <param name="key">The parameter key</param>
+        /// <param name="defaultValue">Default value for this parameter.</param>
+        /// <returns></returns>
+        public float GetParameterWithDefault(string key, float defaultValue)
```

Comment:
Sorry, I might have commented on the wrong thing before. But this is the method I was suggesting we shorten to GetWithDefault()